### ISSUE TITLE
Localization map?

### ISSUE THREAD

**Issue by fabrizioguillaro** [CHELSEA234/HiFi_IFDL#37]
Hello, I was wondering why you are using two different ways to obtain the (binary) localization map.

In HiFi_Net_loc.py, you simply binaryze 'mask_binary' using threshold 0.5:
```
mask1_fea, mask_binary, out0, out1, out2, out3 = SegNet(output, image)
pred_mask = torch.zeros_like(mask_binary)
pred_mask[mask_binary > 0.5] = 1
pred_mask[mask_binary <= 0.5] = 0
```

In HiFi_Net.py, however, in the 'localize' functin, you ignore 'mask_binary' and use 'LOSS_MAP.inference' with 2.3 as threshold:
```
mask1_fea, mask1_binary, out0, out1, out2, out3 = self.SegNet(output, image)
pred_mask, pred_mask_score = self.LOSS_MAP.inference(mask1_fea)
pred_mask_score = pred_mask_score.cpu().numpy()
pred_mask_score[pred_mask_score<2.3] = 0.
pred_mask_score[pred_mask_score>=2.3] = 1.
binary_mask = pred_mask_score[0]  
```

Why are you using two different approaches? Which is the correct one to use?

**Comment by CHELSEA234** [comment]
@fabrizioguillaro sorry for the late response. 
Please use the first method if you want to compare it with the previous method on the IML datasets (e.g., Columbia, CASIA, etc).
If you want to apply this method to detecting digital detection and localization, such from the gan and diffusion, please use the second one.

**Comment by Kelfvin** [comment]
> @fabrizioguillaro sorry for the late response. Please use the first method if you want to compare it with the previous method on the IML datasets (e.g., Columbia, CASIA, etc). If you want to apply this method to detecting digital detection and localization, such from the gan and diffusion, please use the second one.

So which method of generating masks is used for supervision during training? If the supervision used during training and testing is different does it affect the results of the evaluation?
